oml/datasets/audios.py

```diff
@@ -0,0 +1,389 @@
+        extra_data: Optional[Dict[str, Any]] = None,
+        input_tensors_key: str = INPUT_TENSORS_KEY,
+        index_key: str = INDEX_KEY,
+        labels_key: str = LABELS_KEY,
+        sr: int = DEFAULT_SAMPLE_RATE,
+        max_num_seconds: float = DEFAULT_DURATION,
+        num_channels: int = DEFAULT_AUDIO_NUM_CHANNELS,
+        spec_repr_func: Callable[[FloatTensor], FloatTensor] = default_spec_repr_func,
+    ):
+        """
+        Initializes the AudioLabeledDataset.
+
+        Args:
+            df: DataFrame with input data.
+            dataset_root: Base path for audio files.
+            extra_data: Extra data to include in dataset items.
+            input_tensors_key: Key under which audio tensors are stored.
+            index_key: Key for indexing dataset items.
+            labels_key: Key under which labels are stored.
+            sr: Sampling rate of audio files.
+            max_num_seconds: Duration to use from each audio file.
+            num_channels: Number of audio channels.
+            spec_repr_func: Spectral representation extraction function used for visualization.
+        """
+        dataset = AudioBaseDataset(
+            paths=df[PATHS_COLUMN].tolist(),
+            dataset_root=dataset_root,
+            extra_data=extra_data,
+            input_tensors_key=input_tensors_key,
+            index_key=index_key,
+            sr=sr,
+            max_num_seconds=max_num_seconds,
+            num_channels=num_channels,
+            start_times=parse_start_times(df),
+            spec_repr_func=spec_repr_func,
+        )
+        super().__init__(dataset=dataset, df=df, extra_data=extra_data, labels_key=labels_key)
+
+    def visualize(self, item: int, color: TColor) -> np.ndarray:
+        return self._dataset.visualize(item=item, color=color)
+
+    def visualize_with_player(self, item: int, color: TColor) -> str:
+        return self._dataset.visualize_with_player(item=item, color=color)
+
+
+class AudioQueryGalleryDataset(DFQueryGalleryDataset, IVisualizableDataset):
+    """
+    The **non-annotated** dataset of audios having `query`/`gallery` split.
+    To perform `1 vs rest` validation, where a query is evaluated versus the whole validation dataset
+    (except for this exact query), you should mark the item as ``is_query == True`` and ``is_gallery == True``.
+
+    """
+
+    _dataset: AudioBaseDataset
+
+    def __init__(
+        self,
+        df: pd.DataFrame,
+        dataset_root: Optional[Union[str, Path]] = None,
+        extra_data: Optional[Dict[str, Any]] = None,
+        input_tensors_key: str = INPUT_TENSORS_KEY,
+        index_key: str = INDEX_KEY,
+        sr: int = DEFAULT_SAMPLE_RATE,
+        max_num_seconds: float = DEFAULT_DURATION,
+        num_channels: int = DEFAULT_AUDIO_NUM_CHANNELS,
+        spec_repr_func: Callable[[FloatTensor], FloatTensor] = default_spec_repr_func,
+    ):
+        """
+        Initializes the AudioQueryGalleryDataset.
```

Comment:
rm `Initializes the AudioQueryGalleryDataset.`

Reply:
fixed